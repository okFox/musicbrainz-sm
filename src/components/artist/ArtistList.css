.artistList {
    background-color: cornsilk;
    h1 {
    font-family: 'Acme', sans-serif;
}
}