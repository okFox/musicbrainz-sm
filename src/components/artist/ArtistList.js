import React from 'react';
import ArtistItem from './ArtistItem.js';
import styles from './ArtistItem.css';
import PropTypes from 'prop-types';

const ArtistList = ({ artistArray }) => {

  const artistItemElements = artistArray.map(({ artistName, artistId }) => (
    <ArtistItem key={artistId} artistId={artistId} artistName={artistName} />
  ));
  
  return (
    <section className={styles.artistList}>
      <ul>
        {artistItemElements}
      </ul>
    </section>
  );


  
};

ArtistList.propTypes = {
  artistArray: PropTypes.array
};

export default ArtistList;
