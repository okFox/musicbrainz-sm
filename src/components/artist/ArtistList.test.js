import React from 'react';
import { shallow } from 'enzyme';
import ArtistList from './ArtistList';

describe('ArtistList component', () => {
  it('renders ArtistList', () => {
    const wrapper = shallow(<ArtistList artistArray={[{ artistId: '1' }, { artistId: '2' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
