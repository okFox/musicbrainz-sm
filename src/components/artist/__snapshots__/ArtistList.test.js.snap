// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders ArtistList 1`] = `
<section
  className="artistList"
>
  <ul>
    <ArtistItem
      artistId="1"
      key="1"
    />
    <ArtistItem
      artistId="2"
      key="2"
    />
  </ul>
</section>
`;
